pypst/utils.py

```diff
@@ -62,3 +98,33 @@ def render_sequence(arg: Iterable[Any]) -> str:
     Render a sequence of any object supported by `render`.
     """
     return f"({', '.join(render_code(a) for a in arg)})"
+
+
+def render_datetime(arg: date | datetime) -> str:
+    """
+    Render a Python `datetime.date` into a call to the Typst datetime function.
+    """
+    obj = {
+        name: getattr(arg, name)
+        for name in ["year", "month", "day", "hour", "minute", "second"]
+        if hasattr(arg, name)
+    }
+    return f"#datetime{render_mapping(obj)}"
+
+
+def render_timedelta(arg: timedelta) -> str:
+    """
+    Render a Python `datetime.timedelta` into a call to the Typst duration function.
+    """
+    obj = {
+        name: getattr(arg, name)
+        for name in [
+            "seconds",
+            "minutes",
+            "hours",
+            "days",
+            "weeks",
+        ]
+        if hasattr(arg, name)
```

Comment:
`timedelta` only saves `days`, `secods` and `microseconds` internally. There are no `hours`, etc., attributes. ([Docs](https://docs.python.org/3/library/datetime.html#timedelta-objects))

Reply:
Ah, might have been overeager there then. Good catch. Will look that up tomorrow.